spikeinterface/preprocessing/highpass_spatial_filter.py

```diff
@@ -0,0 +1,280 @@
+import numpy as np
+import scipy.signal
+from spikeinterface.preprocessing.basepreprocessor import BasePreprocessor, BasePreprocessorSegment
+from spikeinterface.core.core_tools import define_function_from_class
+
+
+class HighpassSpatialFilter(BasePreprocessor):
+    """
+    Perform destriping with high-pass spatial filtering. Uses
+    the kfilt() function of the International Brain Laboratory.
+
+    Median average filtering, by removing the median of signal across
+    channels, assumes noise is constant across all channels. However,
+    noise have exhibit low-frequency changes across nearby channels.
+
+    This function extends the cut-band from 0 to 0.01 Nyquist to
+    remove these low-frequency changes during destriping.
+
+    Parameters
+    ----------
+
+    n_channel_pad: Number of channels to pad prior to filtering. Channels
+                   are padded with mirroring.
+
+    n_channel_taper: Number of channels to perform cosine tapering on
+                     prior to filtering. If None and n_channel_pad is set,
+                     n_channel_taper will be set to the number of
+                     padded channels. Otherwise, the passed value will be used.
+
+    agc_options: Options for automatic gain control. By default, gain control
```

Comment:
@JoeZiminski What does gain control do exactly? Can you explain a bit more here? From the implementation, it seems to be some sort of scaling, but I wonder whether this should be a separate step, decoupledfrom the actual spatial filter. What do you and @oliche think?

Reply:
Just saw you explained it in the PR text. Fine to leave it here :)